controllers/groupsController.js

```diff
@@ -29,61 +29,71 @@ const summary = async (req, res) => {
 
   try {
     const { uid } = req.params;
-    const postUpdateNewest = [];
-    const keywordListResult = await keywordModel.find({ ownerUid: uid }).exec();
-    const keywordIdList = keywordListResult.map((keyword) => keyword._id);
-    const postCreatedLast = [];
 
-    for await (const id of keywordIdList) {
-      const postResult = await postModel.find({ keywordId: id }).sort({ createdAt: -1 }).exec();
+    const keywordResult = await keywordModel.find({ ownerUid: uid }).sort({ updatedAt: -1 }).exec();
+    const keywordList = keywordResult.map((keyword) => {
+      return {
+        id: keyword._id.toString(),
+        name: keyword.keyword,
+      };
+    });
+
+    const keywordIdList = keywordResult.map((keyword) => keyword._id.toString());
+    let keywordIdNewestHavingPost;
+
+    for await (const keywordId of keywordIdList) {
+      const hasPost = (await postModel.find({ keywordId }).countDocuments().exec()) > 0;
 
-      if (postResult.length > 0) {
-        postCreatedLast.push({
-          keywordId: postResult[0].keywordId,
-          createdAt: postResult[0].createdAt,
-        });
+      if (hasPost) {
+        keywordIdNewestHavingPost = keywordId;
+        break;
       }
     }
+    const keywordUpdatedNewestHavingPost = keywordResult.filter(
+      (keyword) => keywordIdNewestHavingPost === keyword._id.toString()
+    )[0];
 
-    const postCreatedNewest = postCreatedLast?.reduce((prev, curr) => {
-      return new Date(prev.createdAt) <= new Date(curr.createdAt) ? curr : prev;
-    });
-    const dateNewest = postCreatedNewest.createdAt.toString();
+    const lastUpdatedAt = keywordUpdatedNewestHavingPost.updatedAt;
+    const dateNewest = lastUpdatedAt.toString();
```

Comment:
```suggestion
    const lastUpdatedAt = keywordUpdatedNewestHavingPost.updatedAt;
```
`lastUpdatedAt`을 toString()으로 변환하지 않고도 정상적으로 동작하지 않을까라는 생각이 들어서 테스트를 해봤어요!
정상적으로 동작하는 것을 확인했는데, 문자열로 변환하지 않고 바로 사용하는게 코드가 간단해져서 더 좋을 것 같아요!

Reply:
수정 완료했습니다! 처음 `setHours`로 인해 원본 변형이 있었는데, 개별적으로 new Date() 생성자로 활용하면 안전하다는 것 확인했어요 감사합니다 :)